docs/docs/01-ibc/06-channel-upgrades.md

```diff
@@ -11,6 +11,48 @@ slug: /ibc/channel-upgrades
+
+With channel upgradability, applications will be able to implement features such as but not limited to: potentially adding [denom metadata to tokens](https://github.com/cosmos/ibc/discussions/719), or utilizing the [fee middleware](https://github.com/cosmos/ibc/tree/main/spec/app/ics-029-fee-payment), all while maintaining the channels on which they currently operate.
+
+This document outlines the channel upgrade feature, and the multiple steps used in the upgrade process.
+
+## Channel Upgrade Handshake
+
+Channel upgrades will be initialized using a handshake process that is designed to be similar to the standard connection/channel opening handshake.
+
+```go
+type Channel struct {
+  // current state of the channel end
+  State State `protobuf:"varint,1,opt,name=state,proto3,enum=ibc.core.channel.v1.State" json:"state,omitempty"`
+  // whether the channel is ordered or unordered
+  Ordering Order `protobuf:"varint,2,opt,name=ordering,proto3,enum=ibc.core.channel.v1.Order" json:"ordering,omitempty"`
+  // counterparty channel end
+  Counterparty Counterparty `protobuf:"bytes,3,opt,name=counterparty,proto3" json:"counterparty"`
+  // list of connection identifiers, in order, along which packets sent on
+  // this channel will travel
+  ConnectionHops []string `protobuf:"bytes,4,rep,name=connection_hops,json=connectionHops,proto3" json:"connection_hops,omitempty"`
+  // opaque channel version, which is agreed upon during the handshake
+  Version string `protobuf:"bytes,5,opt,name=version,proto3" json:"version,omitempty"`
+  // upgrade sequence indicates the latest upgrade attempt performed by this channel
+  // the value of 0 indicates the channel has never been upgraded
+  UpgradeSequence uint64 `protobuf:"varint,6,opt,name=upgrade_sequence,json=upgradeSequence,proto3" json:"upgrade_sequence,omitempty"`
+}
+```
+
+The version, connection hops, and channel ordering are fields in this channel struct which can be changed. For example, the fee middleware can be added to an application module by updating the version string [shown here](https://github.com/cosmos/ibc-go/blob/995b647381b909e9d6065d6c21004f18fab37f55/modules/apps/29-fee/types/metadata.pb.go#L28). However, although connection hops can change in a channel upgrade, both sides must still be each other's counterparty.
+
+On a high level, successful handshake process for channel upgrades works as follows:
+
+1. The chain initiating the upgrade process will propose a potential upgrade.
+2. If the counterparty agrees with the proposal, it will block sends and begin flushing its channel end. 
```

Comment:
maybe we should be a bit more explicit here about "flushing" since it looks like this is the first time the language is introduced

```suggestion
2. If the counterparty agrees with the proposal, it will block sends and begin flushing any in-flight packets on its channel end. 
```

Reply:
i see what you mean, will add that flushing will be detailed later specifically!